rclcpp/src/rclcpp/event.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rclcpp/event.hpp"
+
+namespace rclcpp
+{
+namespace event
+{
+
+Event::Event()
```

Comment:
I originally called it something like that, but I figured it could be used for other stuff too, since it really is just a generic "event" class. For example, it is currently returned from `node->get_graph_event` but there might be a `node->get_parameter_event` in the future. I don't know if that's the case, but I didn't see any reason it should be less generic, but I'm open to changing it if you like.